tools/README.md

```diff
@@ -38,9 +38,68 @@ for Linux entrypoint) in your `~` directory.
 
 Likewise, `openps` launches powershell.
 
-## runformat
-
-`runformat` will format the c++ code to match our coding style.
+## runformat & runxamlformat
+
+`runxamlformat` will format `.xaml` files to match our coding style. `runformat`
+will format the c++ code (and will also call `runxamlformat`). **`runformat`
+should be called before making a new PR**, to ensure that code is formatted
+correctly. If it isn't, the CI will prevent your PR from merging.
+
+The C++ code is formatted with `clang-format`. Many editors have built-in
+support for automatically running clang-format on save.
+
+Our XAML code is formatted with
+[XamlStyler](https://github.com/Xavalon/XamlStyler). I don't have a good way of
+running this on save, but you can add a got hook to format before committing
+`.xaml` files. To do so, add the following to your `.git/hooks/pre-commit` file:
+
+```sh
+# XAML Styler - xstyler.exe pre-commit Git Hook
+# Documentation: https://github.com/Xavalon/XamlStyler/wiki
+# Originally from https://github.com/Xavalon/XamlStyler/wiki/Git-Hook
+
+# Define path to xstyler.exe
+XSTYLER_PATH="dotnet tool run xstyler --"
+
+# Define path to XAML Styler configuration
+XSTYLER_CONFIG=
+
+echo "Running XAML Styler on committed XAML files"
+git diff --cached --name-only --diff-filter=ACM  | grep -e '\.xaml$' | \

```

Comment:
This will format the wpf xaml files on save, which will make for a huge diff the first time someone touches them. Pre-format them as part of this PR!

Reply:
barf good point